styles/style_option.h

```diff
@@ -0,0 +1,32 @@
+#ifndef STYLES_STYLE_OPTION_H
+#define STYLES_STYLE_OPTION_H
+
+// Usage StyleOption<OPTION, COLORS...>
+// OPTION: a number
+// COLORS: a color or style
+// return value: COLORS
+// color/style option is picked from the specified list of styles/colors
+// based on OPTION value Int<>
+// with Int<0> representing first color/style
+
+template<class F, class... COLORS> 
```

Comment:
I'm wondering if we want to base this on ColorChange<> instead.
It might be useful for more things if there was a transition.

We could make a generic ColorSelect<FUNC, TRANSITION, COLORS...>
Then ColorChange<A, B...> basically becomes ColorSelect<Variation, A, B...>
And StyleOption<A, B....> basically becomes ColorSelect<A, TrInstant, B...>

Reply:
That's a good idea, I can modify, test and resubmit.  Should I keep this PR open or cancel and submit a new one?

Reply:
Up to you, either will work.

Reply:
I renamed style_option to color_select, need your eyes for the HandleFeature addition